### PR TITLE
[Enhancement] [Refactor] Add TaskRunFIFOQueue  to optimize task scheduler

### PR DESCRIPTION
**Pull request by LiShuMing**
## Why I'm doing:
- Pending task runs are stored in `pendingTaskRunMap` and `pendingTaskRunQueue ` separately and it's not good for maintain/interact.
```
    // TODO: Refactor this to find a better way to store the task runs.
    // taskId -> pending TaskRun Queue, for each Task only support 1 running taskRun currently,
    // so the map value is priority queue need to be sorted by priority from large to small
    @SerializedName("pendingTaskRunMap")
    private final Map<Long, Queue<TaskRun>> pendingTaskRunMap = Maps.newConcurrentMap();

    private final TaskRunFIFOQueue pendingTaskRunQueue = new TaskRunFIFOQueue();

```

## What I'm doing:
- Refactor pending task runs into TaskRunFIFOQueue class.
- Use an ordered set for better `poll` and extension.
```
/**
 * A thread safe FIFO queue for pending task runs, it uses an ordered set and hash map to schedule task runs.
 * - ordered set is used to sort task runs by priority and created time.
 * - hash map is used to store task runs by task id.
 */
```

## Further
- Support more task properties, eg: event trigger delay perioid/task priority/task retry attempts.

Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.3
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/scheduler/TaskRun.java [hunk 1]
```diff
@@ -71,18 +72,12 @@ public class TaskRun implements Comparable<TaskRun> {
 
     private TaskRunProcessor processor;
 
-    @SerializedName("status")
     private TaskRunStatus status;
 
-    @SerializedName("type")
     private Constants.TaskType type;
 
-    @SerializedName("executeOption")
```
Comment (ABingHuang) [inline fe/fe-core/src/main/java/com/starrocks/scheduler/TaskRun.java:80]: Is is compatible after upgrade?

Comment (LiShuMing) [reply fe/fe-core/src/main/java/com/starrocks/scheduler/TaskRun.java:80]: No. Those string only used in `inspect_task_runs` and logging.

I found it's confused if displaying too much.

**Review by ABingHuang — APPROVED** [review]

**Review by Astralidea — APPROVED** [review]

**Comment by LiShuMing** [comment]
@mergify backport branch-3.2

**Comment by LiShuMing** [comment]
@mergify backport branch-3.1

**Pull request merged**